### PR TITLE
Add support to pass creation date in feature

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

File: s3/src/main/java/ch/cyberduck/core/s3/S3MetadataFeature.java
Comment (ylangisc) [inline s3/src/main/java/ch/cyberduck/core/s3/S3MetadataFeature.java:1]: Also add creation date. `btime` to be interoperable with rclone. Refer to https://rclone.org/docs/.

File: s3/src/main/java/ch/cyberduck/core/s3/S3WriteFeature.java
Comment (ylangisc) [inline s3/src/main/java/ch/cyberduck/core/s3/S3WriteFeature.java:1]: Add creation date.

File: backblaze/src/main/java/ch/cyberduck/core/b2/B2WriteFeature.java
Comment (ylangisc) [inline backblaze/src/main/java/ch/cyberduck/core/b2/B2WriteFeature.java:1]: To set the creation date we could add a custom `X-Bz-Info-*` header, e.g. `X-Bz-Info-src_creation_date_millis`. Refer to https://www.backblaze.com/apidocs/b2-upload-file

> X-Bz-Info-*
The * part of the header name is replaced with the name of a custom field in the file information stored with the file, and the value is an arbitrary UTF-8 string, percent-encoded. The same info headers sent with the upload will be returned with the download. The header name is case insensitive.

File: box/src/main/java/ch/cyberduck/core/box/BoxWriteFeature.java
Comment (ylangisc) [inline box/src/main/java/ch/cyberduck/core/box/BoxWriteFeature.java:1]: Add `contentCreatedAt`.

File: core/src/main/java/ch/cyberduck/core/transfer/copy/AbstractCopyFilter.java [hunk 1]
```diff
@@ -140,7 +140,7 @@ public TransferStatus prepare(final Path file, final Local n, final TransferStat
             }
         }
         if(options.timestamp) {
```
Comment (ylangisc) [inline core/src/main/java/ch/cyberduck/core/transfer/copy/AbstractCopyFilter.java:142]: Should we also allow extend `options` to allow setting the creation date?

File: core/src/main/java/ch/cyberduck/core/worker/WriteTimestampWorker.java
Comment (ylangisc) [inline core/src/main/java/ch/cyberduck/core/worker/WriteTimestampWorker.java:1]: Not sure if `created` should be part of `equals` and `hashCode` implementation.

File: dracoon/src/main/java/ch/cyberduck/core/sds/SDSDirectS3UploadFeature.java
Comment (ylangisc) [inline dracoon/src/main/java/ch/cyberduck/core/sds/SDSDirectS3UploadFeature.java:1]: Add `timestampCreation`.

File: dracoon/src/main/java/ch/cyberduck/core/sds/SDSUploadService.java
Comment (ylangisc) [inline dracoon/src/main/java/ch/cyberduck/core/sds/SDSUploadService.java:1]: Add creation date.

File: eue/src/main/java/ch/cyberduck/core/eue/EueUploadHelper.java
Comment (ylangisc) [inline eue/src/main/java/ch/cyberduck/core/eue/EueUploadHelper.java:1]: Add creation date.

File: googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveTimestampFeature.java
Comment (ylangisc) [inline googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveTimestampFeature.java:1]: Add `setCreatedDate`.

File: googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java
Comment (ylangisc) [inline googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveWriteFeature.java:1]: Add `setCreatedDate`.

File: googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageCopyFeature.java
Comment (ylangisc) [inline googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageCopyFeature.java:1]: Add `setTimeCreated`.

File: googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageTimestampFeature.java
Comment (ylangisc) [inline googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageTimestampFeature.java:1]: Add `setTimeCreated`.

File: googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageWriteFeature.java
Comment (ylangisc) [inline googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageWriteFeature.java:1]: Add `setTimeCreated`.

File: nextcloud/src/main/java/ch/cyberduck/core/nextcloud/NextcloudWriteFeature.java
Comment (ylangisc) [inline nextcloud/src/main/java/ch/cyberduck/core/nextcloud/NextcloudWriteFeature.java:1]: Add `X-OC-CTime` header.

File: nio/src/main/java/ch/cyberduck/core/nio/LocalTimestampFeature.java
Comment (ylangisc) [inline nio/src/main/java/ch/cyberduck/core/nio/LocalTimestampFeature.java:1]: Add creation date.

File: storegate/src/main/java/ch/cyberduck/core/storegate/StoregateWriteFeature.java [hunk 2]
```diff
@@ -166,8 +166,8 @@ protected String start(final Path file, final TransferStatus status) throws Back
             meta.setParentId(fileid.getFileId(file.getParent()));
             meta.setFileSize(status.getLength() > 0 ? status.getLength() : null);
             meta.setCreated(DateTime.now());
```
Comment (ylangisc) [inline storegate/src/main/java/ch/cyberduck/core/storegate/StoregateWriteFeature.java:168]: Use creation date from `TransferStatus`.

File: webdav/src/main/java/ch/cyberduck/core/dav/DAVTimestampFeature.java
Comment (ylangisc) [inline webdav/src/main/java/ch/cyberduck/core/dav/DAVTimestampFeature.java:1]: Add custom properties for creation date too?

**Review by ylangisc — CHANGES_REQUESTED** [review]

File: googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveTimestampFeature.java
Comment (dkocher) [reply googledrive/src/main/java/ch/cyberduck/core/googledrive/DriveTimestampFeature.java:1]: fieldNotWritable

File: googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageWriteFeature.java
Comment (dkocher) [reply googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageWriteFeature.java:1]: Non-editable metadata in Google Storage

File: googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageTimestampFeature.java
Comment (dkocher) [reply googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageTimestampFeature.java:1]: Non-editable metadata

File: googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageCopyFeature.java
Comment (dkocher) [reply googlestorage/src/main/java/ch/cyberduck/core/googlestorage/GoogleStorageCopyFeature.java:1]: Non-editable metadata

**Comment by ylangisc** [comment]
@dkocher Pending comment: https://github.com/iterate-ch/cyberduck/pull/15101/files/c29d5586b38b31dddfb6f5e79ba54b0b6f38c233#r1324129083

File: webdav/src/main/java/ch/cyberduck/core/dav/DAVTimestampFeature.java
Comment (dkocher) [reply webdav/src/main/java/ch/cyberduck/core/dav/DAVTimestampFeature.java:1]: No priority.

**Pull request merged**